### PR TITLE
Rename openshift-apiserver to just apiserver for clarity

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Renames the deployment of the kube-apiserver for openshift to have the same name as the one when using Kubernetes to make the differentiation a bit more clear, before there was an `apiserver-openshift` deployment for the kube-apiserver in openshift and a `openshift-apiserver` for the openshift-apiserver deployment which is a bit confusing.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

/assign @nikhita

### CODE REVIEW

File: api/pkg/controller/openshift/openshift_controller.go [hunk 1]
```diff
@@ -330,11 +330,11 @@ func (r *Reconciler) syncHeath(ctx context.Context, osData *openshiftData) error
 	}
 
 	healthMapping := map[string]*depInfo{
-		openshiftresources.ApiserverDeploymentName:         {healthStatus: &currentHealth.Apiserver, minReady: 1},
-		openshiftresources.ControllerManagerDeploymentName: {healthStatus: &currentHealth.Controller, minReady: 1},
-		resources.MachineControllerDeploymentName:          {healthStatus: &currentHealth.MachineController, minReady: 1},
-		resources.OpenVPNServerDeploymentName:              {healthStatus: &currentHealth.OpenVPN, minReady: 1},
-		resources.UserClusterControllerDeploymentName:      {healthStatus: &currentHealth.UserClusterControllerManager, minReady: 1},
+		resources.ApiserverDeploymentName:             {healthStatus: &currentHealth.Apiserver, minReady: 1},
+		resources.ControllerManagerDeploymentName:     {healthStatus: &currentHealth.Controller, minReady: 1},
```
Comment (nikhita) [inline api/pkg/controller/openshift/openshift_controller.go:334]: Should this be changed here as well?

https://github.com/kubermatic/kubermatic/blob/c48d09feec1d2553b4e0eb257108a71668737478/api/pkg/controller/openshift/resources/controller_manager_deployment.go#L35

Comment (alvaroaleman) [reply api/pkg/controller/openshift/openshift_controller.go:334]: That is the actual openshift-controller-manager, the kube-controller-manager for openshift is in `resources/kube_controller_manager.go` and named just `controller-manager`.

Maybe we could rename the file from `apiserver_deployment.go` to `kube_apiserver.go` thought, WDYT?

Comment (nikhita) [reply api/pkg/controller/openshift/openshift_controller.go:334]: > Maybe we could rename the file from apiserver_deployment.go to kube_apiserver.go thought, WDYT?

:+1:

**Comment by alvaroaleman** [comment]
@nikhita updated, PTAL

**Review by nikhita — APPROVED** [review]
/lgtm
/approve

**Pull request merged**